core/quantity/src/Formatter/Formatter.ts

```diff
@@ -179,15 +181,15 @@ export class Formatter {
     let componentText = "";
     if (!isLastPart) {
       componentText = Formatter.integerPartToText(compositeValue, spec);
+      if(spec.format.minWidth) { // integerPartToText does not do this padding
+        componentText = this.countAndPad(componentText, spec.format.minWidth);
+      }
     } else {
       componentText = Formatter.formatMagnitude(compositeValue, spec);
     }
 
     if (spec.format.hasFormatTraitSet(FormatTraits.ShowUnitLabel)) {
       componentText = componentText + spec.format.uomSeparator + label;
-    } else {
-      if (!isLastPart)
-        componentText = `${componentText}:`;
```

Comment:
@anmolshres98 @hl662 @ColinKerr @pmconne 
This is one of two breaking fixes that I had to make to the formatting code in order to support this. There are 2 fields which didn't work as in native, but their description on https://www.itwinjs.org/bis/ec/ec-format/ states that the C++ implementation is correct. Still, fixing that here is a breaking change. Let's discuss how to deal with this.
These are the 2 fields:

- "minwidth". The documentation says it will pad left with zeros, but the code actually took the entire formatted string and padded with spaces. In C++ it applies to each component in a composite while the Typescript code just used it on the overall result. In addition, the implementation in ts was buggy, it did effectively nothing, ever. But fixing it will suddenly make it work.
- "Spacer" which is the thing that separates the pieces in a composite format. Somehow, instead of using the spacer, the code always put a ":" between the pieces plus the spacer. That's not what c++ does and the hard coded : (which is the piece of code highlighted by this comment) is also undocumented and I believe unwanted.

For now I just went ahead and fixed the logic, not caring about the break.....

Reply:
I'm okay with both of these changes, seeing as its inherently a fix. The bad behavior was undetectable, or not big enough to warrant for product teams to raise an issue about it. Since it could cause users to assume this bug was intended for a long time, it won't hurt to document these bug fixes in `NextVer.md` though. Anyone else have any thoughts?

Reply:
closing the conversation. Nobody seems to object.